gen/calc.py

```diff
@@ -547,6 +547,37 @@ def validate_exhibitor_storage_master_discovery(master_discovery, exhibitor_stor
             "`master_http_load_balancer` then exhibitor_storage_backend must not be static."
 
 
+def calculate_adminrouter_tls_version_override(adminrouter_tls_1_0_enabled, adminrouter_tls_1_1_enabled,
+                                               adminrouter_tls_1_2_enabled):
+    tls_versions = list()
+    if adminrouter_tls_1_0_enabled == 'true':
+        tls_versions.append('TLSv1')
+
+    if adminrouter_tls_1_1_enabled == 'true':
+        tls_versions.append('TLSv1.1')
+
+    if adminrouter_tls_1_2_enabled == 'true':
+        tls_versions.append('TLSv1.2')
+
+    tls_version_string = " ".join(tls_versions)
+    return tls_version_string
+
+
+def calculate_adminrouter_tls_cipher_override(adminrouter_tls_cipher_suite):
+    if adminrouter_tls_cipher_suite != '':
+        return 'true'
+    else:
+        return 'false'
+
+
+def validate_adminrouter_tls_version_override(adminrouter_tls_version_override, adminrouter_tls_1_0_enabled,
```

Comment:
I believe that `adminrouter_tls_1_0_enabled`, `adminrouter_tls_1_1_enabled` and `adminrouter_tls_1_2_enabled` are string variables with `'true'` and `'false'` content. `any()` function returns false only on empty array or all elements of the array are boolean `False`.

```
assert any(['false', 'false']) == True
```

vs

```
assert any([False, False]) == False
```
We need to fix this logic and add appropriate unit test that checks if the message is issued under expected conditions.

Reply:
I forgot that I need an in clause in any when using strings - not sure why we don't use booleans in calc.py generally, I just followed the other examples.  Will fix in a follow up commit.